plugin/documents.py

```diff
@@ -492,9 +504,35 @@ def _when_selection_remains_stable_async(self, f: Callable[[], None], r: sublime
 
     def _register_async(self) -> None:
         file_name = self.view.file_name()
-        if file_name:
-            self._file_name = file_name
-            self.manager.register_listener_async(self)
+        if not file_name:
+            debug("listener", self, "has  no file name")
+            return
+        self._file_name = file_name
+        buf = self.view.buffer()
+        if not buf:
+            debug("not tracking bufferless view", self.view.id())
+            return
+        text_change_listener = TextChangeListener.ids_to_listeners.get(buf.buffer_id)
+        if not text_change_listener:
+            debug("couldn't find a text change listener for listener", self)
+            return
+        text_change_listener.view_listeners.add(self)
+        self._text_change_listener = ref(text_change_listener)
+        self.manager.register_listener_async(self)
+        views = buf.views()
+        if not isinstance(views, list):
+            debug("skipping clone checks listener", self)
+            return
+        self_id = self.view.id()
+        for view in views:
+            view_id = view.id()
+            if view_id == self_id:
+                continue
+            listeners = list(sublime_plugin.view_event_listeners[view_id])
+            for listener in listeners:
+                if isinstance(listener, DocumentSyncListener):
+                    debug("registering clone listener", listener)
```

Comment:
It's not necessarily a clone. The current one might be a clone.

Reply:
Is the current wording OK?